Use GitHub Action to fetch translations from Transifex daily

gouenji-shuuya opened the pull request:
Everyday around 00:00 UTC (+/- some offset), this action will fetch translations from Transifex and commit them if there are any changes.

This action will also be triggered by:
- Updates to Python files
- Updates to this file.
- Manually from actions page.

The following must be set up as secrets in the repository:
- ~~SOME_STRING: Anything. Changing this invalidates pip cache.~~
- TRANSIFEX_API_TOKEN: Self-explanatory.

akprasad commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,69 @@
+                run: |
+                    if [[ $(git status -s) ]]; then
+                        git config user.name github-actions
+                        git config user.email github-actions[bot]@users.noreply.github.com
```
Let's use `git config user.email 41898282+github-actions[bot]@users.noreply.github.com` used so that we get the bot icon as well:

https://github.com/orgs/community/discussions/26560

gouenji-shuuya replied:
The email used also shows it (that's why I used it).

akprasad replied:
ah, wonderful! good to know

akprasad commented on .github/workflows/actions.yml:
```diff
@@ -0,0 +1,69 @@
+            # https://stackoverflow.com/a/62639424
+            - name: Cache virtual environment
+                uses: actions/cache@v3
+                id: cache-venv
```
rather than this, can we simply cache on the `requirements.txt` file?

gouenji-shuuya replied:
On every run it will download all the dependencies, which is a waste of time. This caches those dependencies.

`requirements.txt` is already in the repo statically, so caching it doesn't make sense.

gouenji-shuuya replied:
Actually, looking at it again, since we run only once a day, even a minute extra won't really matter.

Maybe I should remove this.

akprasad replied:
What I mean is that the cache key can be determined by the hash of `requirements.txt` so that if that file changes, GitHub will know to re-fetch. Actually, I think this might be default behavior:

https://github.com/actions/setup-python#caching-packages-dependencies

gouenji-shuuya replied:
Oh, I didn't know they had added support for pip cache natively (https://github.com/actions/setup-python/pull/266). Thanks.

gouenji-shuuya committed:
Use GitHub Action to fetch translations from Transifex daily

Everyday around 00:00 UTC (+/- some offset), this action will fetch
translations from Transifex and commit them if there are any changes.

This action will also be triggered by:
- Updates to Python files
- Updates to this file.
- Manually from actions page.

TRANSIFEX_API_TOKEN must be set up as a secret in the repository.

akprasad commented:
Awesome! Very happy about this -- I'll merge it in then kick off a manual run to test.

gouenji-shuuya commented:
Glad to help!
